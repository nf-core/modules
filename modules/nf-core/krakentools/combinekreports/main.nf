process KRAKENTOOLS_COMBINEKREPORTS {
    label 'process_single'

    conda "bioconda::krakentools=1.2"
    container "${ workflow.containerEngine == 'singularity' && !task.ext.singularity_pull_docker_container ?
        'https://depot.galaxyproject.org/singularity/krakentools:1.2--pyh5e36f6f_0':
        'quay.io/biocontainers/krakentools:1.2--pyh5e36f6f_0' }"

    input:
    tuple val(meta), path(kreports)

    output:
    tuple val(meta), path("*.txt"), emit: txt
    path "versions.yml", emit: versions

    when:
    task.ext.when == null || task.ext.when

    script:
    def args = task.ext.args ?: ''
    prefix = task.ext.prefix ?: "${meta.id}"
    def VERSION = '1.2' // WARN: Version information not provided by tool on CLI. Please update this string when bumping container versions.
    """
    combine_kreports.py \\
        -r ${kreports} \\
        -o ${prefix}.txt \\
        ${args}

    cat <<-END_VERSIONS > versions.yml
    "${task.process}":
        combine_kreports.py: ${VERSION}
    END_VERSIONS
    """
}
