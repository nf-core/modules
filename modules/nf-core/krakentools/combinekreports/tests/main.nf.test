nextflow_process {

    name "Test Process KRAKENTOOLS_COMBINEKREPORTS"
    script "../main.nf"
    process "KRAKENTOOLS_COMBINEKREPORTS"

    tag "modules"
    tag "modules_nfcore"
    tag "krakentools"
    tag "krakentools/combinekreports"

    test ("sarscov2 - metagenome - kraken report") {

        when {
            process {
                """
                input[0] = Channel.of([
                   [id: 'test'],
                   file(params.modules_testdata_base_path + 'genomics/sarscov2/metagenome/test_1.kraken2.report.txt', checkIfExists: true)
                ])
                """
            }
        }

        then {
            assertAll(
                { assert process.success },
                { assert snapshot ( process.out ).match() }
            )
        }

    }

    test ("sarscov2 - metagenome - kraken report - stub") {

        options '-stub'
        when {
            process {
                """
                input[0] = Channel.of([
                   [id: 'test'],
                   file(params.modules_testdata_base_path + 'genomics/sarscov2/metagenome/test_1.kraken2.report.txt', checkIfExists: true)
                ])
                """
            }
        }

        then {
            assertAll(
                { assert process.success },
                { assert snapshot ( process.out ).match() }
            )
        }
    }
}
