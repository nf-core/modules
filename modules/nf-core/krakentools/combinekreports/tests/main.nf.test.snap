{
    "sarscov2 - metagenome - kraken report": {
        "content": [
            {
                "0": [
                    [
                        {
                            "id": "test"
                        },
                        "test.txt:md5,481c4b987b922979baf74f2ce8d88b02"
                    ]
                ],
                "1": [
                    "versions.yml:md5,a907962abbd9f5dcd34c9e8a4e6100d0"
                ],
                "txt": [
                    [
                        {
                            "id": "test"
                        },
                        "test.txt:md5,481c4b987b922979baf74f2ce8d88b02"
                    ]
                ],
                "versions": [
                    "versions.yml:md5,a907962abbd9f5dcd34c9e8a4e6100d0"
                ]
            }
        ],
        "meta": {
            "nf-test": "0.8.4",
            "nextflow": "23.04.1"
        },
        "timestamp": "2024-08-07T09:12:42.382189"
    },
    "sarscov2 - metagenome - kraken report - stub": {
        "content": [
            {
                "0": [
                    [
                        {
                            "id": "test"
                        },
                        "test.txt:md5,d41d8cd98f00b204e9800998ecf8427e"
                    ]
                ],
                "1": [
                    "versions.yml:md5,a907962abbd9f5dcd34c9e8a4e6100d0"
                ],
                "txt": [
                    [
                        {
                            "id": "test"
                        },
                        "test.txt:md5,d41d8cd98f00b204e9800998ecf8427e"
                    ]
                ],
                "versions": [
                    "versions.yml:md5,a907962abbd9f5dcd34c9e8a4e6100d0"
                ]
            }
        ],
        "meta": {
            "nf-test": "0.8.4",
            "nextflow": "23.04.1"
        },
        "timestamp": "2024-08-07T09:12:46.705759"
    }
}
